utils/s3/client.hh

```diff
@@ -30,10 +31,43 @@ namespace s3 {
 
 using s3_clock = std::chrono::steady_clock;
 
-struct range {
-    uint64_t off;
-    size_t len;
+class range {
+    friend struct fmt::formatter<range>;
+
+public:
+    range() noexcept = delete;
+    constexpr range(uint64_t offset, uint64_t len) noexcept : _offset(offset), _length(len) {
+        uint64_t t;
+        SCYLLA_ASSERT(!__builtin_add_overflow(_offset, len, &t));
+        SCYLLA_ASSERT(_offset + len <= _max_object_size);
```

Comment:
Assertion looks too severe

Reply:
Do we want to switch to throwing exception instead?